### PR TITLE
Change interpolation in apply_transform() from nearest to bilinear.

### PR DESCRIPTION
**Pull request by ozabluda**
Before https://github.com/keras-team/keras/pull/2446, interpolation was inconsistently sometimes nearest, sometimes bicubic. AFAIK, in other libraries and tools default interpolation is never 'nearest'. Sometimes it's bicubic, sometimes it's bilinear. For Keras bilinear is better because it's much faster, and is something a single (de)convolutional layer can learn (i.e. it is, basically a single hardcoded non-trainable (de)convolutional layer. Which is not true of nearest, even with pooling.

Also see `load_img()` https://github.com/keras-team/keras/pull/8435

### CODE REVIEW

**Comment by fchollet** [comment]
This is also a potentially breaking change.

**Comment by ozabluda** [comment]
Yes, but it doesn't affect any pretrained models. Current default is highly unexpected, since we have fractional shifts, rotations, zooms and whatnot, and probably makes things worse vast majority of the time. I actually spent some time looking at `'nearest'` interpolation when I was dealing with binarized images, which had to stay binarized after data augmentation. It sucks (augmented data doesn't correspond to the original). For that reason, I am not even proposing adding `order` parameter, since it's so exotic. A slightly less exotic case is `order=3` for, say, superresolution autoencoder, but even than moved to `order=1` AFAIK, probably for the reasons in my description above.

But we could introduce `order` parameter with default `None` and print warning that default changed, but I don't think anybody would care (but it would be useful mostly for debugging). Then revert to default `order=1` when we have a an API-breaking release. Or we can keep `None` so that sometimes we do indeed default to `order=0` when, say, we have integer shifts only (something like `random_crop()`, for which I am still designing of a good API, and, maybe, it would work just fine with `order=1`.

**Comment by Dref360** [comment]
This would break existing code for segmentation DA.

**Comment by ozabluda** [comment]
what is "segmentation DA"?

**Comment by Dref360** [comment]
Sorry,
Data augmentation on segmentation map.
If you do a bilinear interpolation, you would corrupt the data. For example, a pixel with class "4" could become "4.5" if its neighbors are not of the same class.

**Comment by ozabluda** [comment]
I thought segmentation DA currently doesn't really work anyway: search for "segmentation" in  https://github.com/keras-team/keras/issues/3338. Also see https://github.com/keras-team/keras/issues/6538. Also 2D softmax was merged only a couple of weeks ago. But it is a valid use case for nearest, thank you.

**Comment by ozabluda** [comment]
I knew it! Use random linear interpolation for data & **labels** to improve stuff

mixup: Beyond Empirical Risk Minimization (1017)
https://arxiv.org/abs/1710.09412v1
